### DISCUSSION TITLE
Видимость полей в канбане

### DISCUSSION THREAD

**Discussion by papaoleg** [totumonline/totum-ru-issues-and-discussions#485] [ОШБК!] [answered]
Подскажите, есть ли решение проблемы: в таблице есть поля, которые недоступны определенным ролям. Когда вид таблицей, то проблем нет, все видят то, что должны. Но когда переключаешься в вид панелями (канбан), то таблица вовсе не выводится. Получается, что вид канбаном работает только тогда, когда все поля доступны всем пользователям?
Можно ли как-то скрыть поля по ролям при включении канбана?
Заранее благодарю за ответ.

**Comment by ttm-alexeypolunin** [comment]
Это ошибка Тотума. Поставил в очередь на исправление. Но скорее всего уже в 5 версии: https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/480#discussioncomment-7753277

**Reply by papaoleg** [reply]
Благодарю. А есть примерные сроки перехода на пятую версию?

**Reply by papaoleg** [reply]
Пока решил проблему через сетформат:
f1=: setFormat(condition: $#nu = $nu; text: "Заблокировано"; textasvalue: true)
Ну и плюс отключение лога для всех, кому не надо...

**Reply by ttm-alexeypolunin** [reply]
> А есть примерные сроки перехода на пятую версию?

Я рассчитываю на первую половину января.

**Reply by ttm-alexeypolunin** [reply]
Мы полезли исправлять и у нас не воспроизводится. Непонятно, мы починили паралельно с другой ошибкой или просто не можем воспроизвести. Посмотрите, какая ошибка в консоли браузера в этом случае — скажите нам если есть возможность.

**Reply by papaoleg** [reply]
Ошибок никаких не наблюдаю. Или не знаю, где наблюдать, логи одинаковые при любом раскладе у любого пользователя. Вот так таблица выглядит для роли со всеми доступами:
![2024-01-12_18-16-39](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/143928075/99156162-8c0e-4286-afc8-b282db1ce02f)
А так для того (роль Работник склада), у кого выключено поле Сумма (при этом табличный вариант отображается правильно):
![2024-01-12_18-17-06](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/143928075/a2b7aefd-02ba-44de-888e-c30004451efd)
Настройка поля Сумма:
![2024-01-12_18-21-21](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/143928075/9a992816-71d4-4738-b019-40c5581a9252)

**Reply by ttm-alexeypolunin** [reply]
Если это Хром то Правой кнопкой в пустом простр страницы — нижнее Посмотреть код — Консоль — Текст ошибки будет внизу справа:

<img width="1724" alt="Снимок экрана 2024-01-12 в 16 16 05" src="https://github.com/totumonline/totum-ru-issues-and-discussions/assets/55755565/ef2a29df-3024-453a-ba68-cea7dcf32c71">

Там должно быть что-то красное

**Reply by papaoleg** [reply]
Uncaught TypeError: this.tableRow.panels_view.fields.forEach is not a function
    at App.pcTableMain.e (main.js?v=c898c3b:3:189694)
    at main.js?v=c898c3b:3:198098
    at Array.forEach (<anonymous>)
    at main.js?v=c898c3b:3:198013
    at Array.forEach (<anonymous>)
    at App.pcTableMain.i (main.js?v=c898c3b:3:197228)
    at App.pcTableMain.n (main.js?v=c898c3b:3:200186)
    at App.pcTableMain.render (main.js?v=c898c3b:3:222160)
    at new App.pcTableMain (main.js?v=c898c3b:1:206893)
    at HTMLDocument.<anonymous> (172:99:9)

**Reply by ttm-alexeypolunin** [reply, marked as answer]
Исправили V5
